activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -396,7 +404,7 @@ def indexes(table_name, name = nil) #:nodoc:
             end
 
             indexes.last.columns << row[:Column_name]
-            indexes.last.lengths.merge!(row[:Column_name] => row[:Sub_part].to_i) if row[:Sub_part]
+            indexes.last.lengths.merge!(row[:Column_name] => row[:Sub_part].to_i) if row[:Sub_part] && mysql_index_type != :spatial
```

Comment:
Mysql returns non-null `Sub_part` for spatial index, but does not allow to set it, thus previous behaviour resulted in a schema dump that could not be loaded back.

https://dev.mysql.com/doc/refman/5.7/en/create-index.html :

> Spatial indexes (created using SPATIAL INDEX) have these characteristics:
> ...
> - Column prefix lengths are prohibited. The full width of each column is indexed.